test/common/mongoose_helper.erl

```diff
@@ -499,6 +500,22 @@ restart_listener(Spec, Listener) ->
     rpc(Spec, mongoose_listener, stop_listener, [Listener]),
     rpc(Spec, mongoose_listener, start_listener, [Listener]).
 
+change_listener_idle_timeout(ListenerPort, Timeout) ->
+    ListenerConfig = get_listener_config_by_port(ListenerPort),
+    #{protocol := ProtocolOpts} = ListenerConfig,
+    % Chage the default idle_timeout to 5s to test if sse will override it
+    NewConfig = ListenerConfig#{protocol => ProtocolOpts#{idle_timeout => Timeout}},
+    restart_listener(mim(), NewConfig).
+
+restart_listener_on_port(ListenerPort) ->
+    ListenerConfig = get_listener_config_by_port(ListenerPort),
+    restart_listener(mim(), ListenerConfig).
+
+get_listener_config_by_port(ListenerPort) ->
```

Comment:
You can just use `get_listeners/2` instead, and this is what other tests are doing. The reason we don't get them by port is because port numbers can change.

Reply:
I used `get_listeners/2` but I had to change it to be recursive (so we don't need to pass the whole handler map but only the keys that we are interested). Please have a look at the new implementation.

Reply:
It's fine for me. Other tests do it differently (they iterate over handlers and modify selecte ones), so there's a bit of inconsistency, but this approach is the most extensible and reusable.